src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -275,6 +285,24 @@ grpc_error* grpc_tcp_client_prepare_fd(const grpc_channel_args* channel_args,
   return GRPC_ERROR_NONE;
 }
 
+/* Fills in dest with a new string containing the human readable
+ * source address of fd. If source address extraction fails for any reason,
+ * this then fills in dest with "source_address_extraction_failed". */
+static void extract_source_ip(char** dest, int fd) {
```

Comment:
following C++ TOTW 176 - Prefer return values to output parameters,
this should be -
static char * <func_name> (int fd)

Reply:
done